### PR TITLE
[SYCL-MLIR]: Update supported functions

### PR DESCRIPTION
**Pull request by whitneywhtsang**
We decided to create a whitelist of SYCL functions to emit instead just the declaration. 
Eventually, this list should be removed.

As we are focusing on the single_task with one dimensional write buffer scenario, update the list to just the SYCL functions that are required by that scenario .

Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

**Review by etiotto — DISMISSED** [review]

**Comment by whitneywhtsang** [comment]
https://github.com/intel/llvm/pull/6756 adds support to the two commented out SYCL functions.

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 1]
```diff
@@ -3664,7 +3663,7 @@ ValueCategory MLIRScanner::VisitCastExpr(CastExpr *E) {
     }
     assert(se.val);
     auto Derived =
-         (E->isLValue() || E->isXValue())
+        (E->isLValue() || E->isXValue())
```
Comment (arnamoy10) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:3666]: Unrelated change?

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 2]
```diff
@@ -116,32 +116,31 @@ MLIRScanner::MLIRScanner(MLIRASTConsumer &Glob,
...
+  supportedFuncs.insert(
+      "_ZN4sycl3_V18accessorIiLi1ELNS0_6access4modeE1025ELNS2_"
+      "6targetE2014ELNS2_11placeholderE0ENS0_3ext6oneapi22accessor_property_"
+      "listIJEEEE6__initEPU3AS1iNS0_5rangeILi1EEESE_NS0_2idILi1EEE");
+  supportedFuncs.insert("_ZN4sycl3_V16detail5arrayILi1EEC1ILi1EEENSt9enable_"
+                        "ifIXeqT_Li1EEmE4typeE");
+  supportedFuncs.insert("_ZN4sycl3_V16detail5arrayILi1EEC1ERKS3_");
+  supportedFuncs.insert("_ZN4sycl3_V12idILi1EEC1Ev");
+  // TODO: Add support to the commented out functions.
+  // supportedFuncs.insert("_ZN4sycl3_V12idILi1EEC1ERKS2_");
```
Comment (arnamoy10) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:140]: Can you please add to the comment why you commented out these functions?

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 3]
```diff
@@ -116,32 +116,31 @@ MLIRScanner::MLIRScanner(MLIRASTConsumer &Glob,
...
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeEmm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi2EEC1ILi2EEENSt9enable_ifIXeqT_Li2EEmE4typeEmm");
-  supportedCons.insert(
-      "_ZN2cl4sycl2idILi3EEC1ILi3EEENSt9enable_ifIXeqT_Li3EEmE4typeEmm");
-  supportedCons.insert("_ZN2cl4sycl6detail5arrayILi1EEC1ILi1EEENSt9enable_"
-                       "ifIXeqT_Li1EEmE4typeE");
+void MLIRScanner::initSupportedFunctions() {
+  // Functions needed for single_task with one dimensional write buffer.
```
Comment (arnamoy10) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:120]: May be elaborate why this functions are needed in this comment too (copy from the PR description)

**Review by arnamoy10 — CHANGES_REQUESTED** [review]

File: polygeist/tools/cgeist/Lib/clang-mlir.cc (hunk 1 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Lib/clang-mlir.cc:3666]: Right, just clang-format. Let's have it in, so we can clang-format the whole file in the future.

File: polygeist/tools/cgeist/Lib/clang-mlir.cc (hunk 3 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Lib/clang-mlir.cc:120]: Added to the declaration.

File: polygeist/tools/cgeist/Lib/clang-mlir.cc (hunk 2 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Lib/clang-mlir.cc:142]: Added comment. These functions will be uncommented in https://github.com/intel/llvm/pull/6756.

File: polygeist/tools/cgeist/Lib/clang-mlir.cc (hunk 1 above)
Comment (arnamoy10) [reply polygeist/tools/cgeist/Lib/clang-mlir.cc:3667]: If we are planning to clang-format the file, I would prefer to be a separate patch.  Let's not pollute git history.

**Review by arnamoy10 — APPROVED** [review]
LGTM
